src/components/ChatApp/MessageSection/MessageSection.react.js

```diff
@@ -184,32 +74,70 @@ class MessageSection extends Component {
 
   constructor(props) {
     super(props);
-    this.state = getStateFromStores();
-    this.customTheme = {
-      header: this.state.header.substring(1),
-      pane: this.state.pane.substring(1),
-      body: this.state.body.substring(1),
-      composer: this.state.composer.substring(1),
-      textarea: this.state.textarea.substring(1),
-      button: this.state.button.substring(1),
+    this.state = {
+      player: [],
+      tour: true,
+      search: false,
+      showLoading: false,
+      showScrollBottom: false,
+      showScrollTop: false,
+      searchState: {
+        markedMessagesByID: {},
+        markedIDs: [],
+        markedIndices: [],
+        scrollLimit: 0,
+        scrollIndex: -1,
+        scrollID: null,
+        caseSensitive: false,
+        searchIndex: 0,
+        searchText: '',
+      },
     };
+    this.messageStoreLength = 0;
+  }
+
+  componentDidMount() {
+    this.scrollToBottom();
+  }
+
+  componentDidUpdate() {
+    const { search, searchState } = this.state;
+    const { messages } = this.props;
+    if (search) {
+      if (searchState.scrollIndex === -1 || searchState.scrollIndex === null) {
+        this.scrollToBottom();
+      } else {
+        const markedIDs = searchState.markedIDs;
+        const markedIndices = searchState.markedIndices;
+        const limit = searchState.scrollLimit;
+        const ul = this.messageList;
+        if (markedIDs && ul && limit > 0) {
+          const currentID = markedIndices[searchState.scrollIndex];
+          this.scrollarea.view.childNodes[currentID].scrollIntoView();
+        }
+      }
+    } else if (messages.length !== this.messageStoreLength) {
```

Comment:
This is the only workaround I could think of to trigger scrollToBottom when a new message is typed or a reply comes and not to fire it just because the scroll view re-rendered.